assets/src/components/swingsView.tsx

```diff
@@ -73,8 +89,30 @@ const SwingsView = (): ReactElement<HTMLElement> => {
 
   const mobileMenuClass = mobileMenuIsOpen ? "blurred-mobile" : ""
 
+  useLayoutEffect(() => {
+    const element = elementRef.current
+
+    // in addition to other criteria, wait until swings are populated so that the
+    // element is tall enough to scroll
+    if (isInitialRender && element && swings) {
```

Comment:
The swings shown are for the current service date, so unless "show past swings" is enabled the number of rows will slowly go down over a given day (the total number of past + future swings will only change when the service day rolls over, though). That does raise some interesting questions, though. The main use case for this is when the user is quickly going back and forth between views, without much time passing, but the behavior could be strange if the user comes back after an hour or so. There is a case to be made for somehow remembering which actual entry the view was scrolled to instead of `scrollTop`, but that would introduce a whole lot of complications. My thought was to make this consistent with how the corresponding functionality for the notifications drawer works, and that similarly just remembers the position as opposed to the particular entry. Still, it's something I can discuss with design + product when we chat.